core/src/main/java/org/commonjava/indy/core/expire/ScheduleManager.java

```diff
@@ -699,34 +699,54 @@ private synchronized void removeCache( final ScheduleKey cacheKey )
     @CacheEntryCreated
     public void scheduled( final CacheEntryCreatedEvent<ScheduleKey, Map> e )
     {
-        final ScheduleKey expiredKey = e.getKey();
-        final Map expiredContent = e.getValue();
-        if ( expiredKey != null && expiredContent != null )
+        if ( e == null )
         {
-            logger.debug( "Expiration Created: {}", expiredKey );
-            final String type = (String) expiredContent.get( ScheduleManager.JOB_TYPE );
-            final String data = (String) expiredContent.get( ScheduleManager.PAYLOAD );
-            eventDispatcher.fire( new SchedulerScheduleEvent( type, data ) );
+            throw new IllegalArgumentException( "[FATAL]The infinispan cache created event for indy schedule manager is null." );
+        }
+
+        if ( !e.isPre() )
+        {
+            final ScheduleKey expiredKey = e.getKey();
+            final Map expiredContent = e.getValue();
+            if ( expiredKey != null && expiredContent != null )
+            {
+                logger.debug( "Expiration Created: {}", expiredKey );
+                final String type = (String) expiredContent.get( ScheduleManager.JOB_TYPE );
+                final String data = (String) expiredContent.get( ScheduleManager.PAYLOAD );
+                eventDispatcher.fire( new SchedulerScheduleEvent( type, data ) );
+            }
         }
     }
 
     @CacheEntryExpired
     public void expired( CacheEntryExpiredEvent<ScheduleKey, Map> e )
     {
-        final ScheduleKey expiredKey = e.getKey();
-        final Map expiredContent = e.getValue();
-        if ( expiredKey != null && expiredContent != null )
+        if ( e == null )
+        {
+            throw new IllegalArgumentException( "[FATAL]The infinispan cache expired event for indy schedule manager is null." );
```

Comment:
Before we release this, we need to change this into a logged error, not an IllegalArgumentException...